### ISSUE TITLE
How is this different from inline styles?

### ISSUE THREAD

**Issue by colmtuite** [tailwindcss/discuss#3]
im sorry i couldn't resist

**Comment by adamwathan** [comment]
![rockeyeroll](https://user-images.githubusercontent.com/4323180/32292336-334d8cd4-bf16-11e7-8301-306ace22c978.gif)

**Comment by jrmorales** [comment]
I recommend you listen to this: 
[⋅⋅* 75: Diana Mounter - Design Systems and Utility Classes at GitHub.](http://www.fullstackradio.com/75)

**Comment by reinink** [comment]
![puking](https://user-images.githubusercontent.com/882133/32292636-20efb07a-bf17-11e7-8809-b387901f1929.gif)

**Comment by OwenMelbz** [comment]
“Dynamic inline styles”

**Comment by mkarnicki** [comment]
Both of those gifs are spot on you guys 😂

**Comment by nuxdie** [comment]
> I recommend you listen to this:
> 
> * [75: Diana Mounter - Design Systems and Utility Classes at GitHub.](http://www.fullstackradio.com/75)

@adamwathan so if I understood correctly the main benefit of adding a CSS framework like tailwindcss is in that you're essentially adding another layer of abstraction between raw CSS and somewhat "traditional" and "complex" (like BEM) CSS approach. In this way you can define your own "DSL" for the styling that makes sense to you and your product, putting a design system on top of raw CSS and tweak it to your taste until you arrive at something you like. After that, you just use it for all styling of more complex components, blocks, and pages. In that case, I would suggest that a perfect fit for your particular website would be produced by manually developing such a system and not adopting an existing one, such as tailwindcss. Please correct me if I'm wrong here, but the only benefit of a project such as tailwindcss is to compromise some time and money for your project in exchange for a readymade solution that still requires a deep understanding of ideas and tradeoffs put into it by its authors.

**Comment by OwenMelbz** [comment]
@nuxdie I’m not answering for Adam, but unless I misunderstood - I wouldn’t say Tailwind is a compromise in anyway shape or form.

IF you have a bespoke design that has no loyalties to using prebuilt components like bootstrap then TW allows you to generate your own framework for your application. 

It has no opinion of itself - merely starter classes as examples which if they happen to work for you great! Otherwise change :) - we NEVER use the default config as it doesn’t cater for our needs, however we can add our rules in from the design team to create a consistent set of styles which the whole site will adhere to.

In my opinion tailwind is not a framework. it’s nothing like bootstrap/bulma/foundation/etc.

It’s a tool to generate your own bespoke framework based off your design requirements.

Something like basecss would in my opinion be a utility class framework

**Comment by adamwathan** [comment]
> It’s a tool to generate your own bespoke framework based off your design requirements.

Yes this is right 👍 Tailwind was designed from the beginning to be hyper-customizable specifically because all of the existing solutions were not designed with deep customization in mind. We provide a sensible default configuration file to make it easy for people to try Tailwind, but for real projects the idea is that you'd take whatever design system your design team has put together and encode it into your own `tailwind.js` configuration file.

**Comment by nuxdie** [comment]
@adamwathan @OwenMelbz thanks for the clarification!
